oak-jcr/src/test/java/org/apache/jackrabbit/oak/jcr/session/SessionLogoutTest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jackrabbit.oak.jcr.session;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+
+import javax.jcr.Session;
+
+import org.apache.jackrabbit.test.AbstractJCRTest;
+
+public class SessionLogoutTest extends AbstractJCRTest {
+
+    public void testConcurrentSessionLogout() throws Exception {
+        Session s = getHelper().getReadWriteSession();
+        List<Exception> exceptions = new ArrayList<>();
```

Comment:
(Just a note to self: I was first wondering whether you'd need a concurrent version of List (eg ConcurrentLinkedQueue) - but Thread.join indeed, and somewhat obviously, honours the ['happens-before order' ](https://docs.oracle.com/javase/specs/jls/se7/html/jls-17.html#jls-17.4.5), so this works perfectly fine.)

Reply:
Hmm, right. But the List must still be thread-safe, which is currently not the case. Multiple threads may add exceptions to the list concurrently in the failure case. I'm going to change that.